src/StructuredLogViewer.Core/NodeQueryMatcher.cs

```diff
@@ -22,6 +21,8 @@ public class NodeQueryMatcher
         private bool searchInValue { get; set; }
         private string nameToSearch { get; set; }
         private string valueToSearch { get; set; }
+        private HashSet<int> UnderNodes { get; set; } = new HashSet<int>();

```

Comment:
I think these will need to be converted to `List<NodeQueryMatcher>` to allow for arbitrary expressions in parentheses

Reply:
sorry I didn't see this tipp $task Csc under($project Foo) and didn't imagine this way to search.  I will make it better.:-)

Reply:
I've changed the logik to allow arbitrary expression in under() notunder(), need to look at this together